input/dir/dir_input.go

```diff
@@ -0,0 +1,55 @@
+package dir
+
+import (
+	"crypto/sha512"
+	"hash"
+	"os"
+
+	"github.com/spacemonkeygo/errors"
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/input"
+	"polydawn.net/repeatr/lib/fshash"
+)
+
+const Type = "dir"
+
+// interface assertion
+var _ input.Input = &Input{}
+
+type Input struct {
+	spec          def.Input
+	hasherFactory func() hash.Hash
+}
+
+func New(spec def.Input) *Input {
+	if spec.Type != Type {
+		panic(errors.ProgrammerError.New("This input implementation supports definitions of type %q, not %q", Type, spec.Type))
```

Comment:
Think of it like the panic in the `default` of a switch handling an enum.  Actually hitting this would be a compile-time fuckup, yes, but I for one make mistakes like that all the time, and that's also why it's in an error hierarchy that's never supposed to be caught.